### DISCUSSION TITLE
set version Ubuntu-20.04 No distribution with supplied name

### DISCUSSION THREAD

**Discussion by donnrri** [microsoft/WSL#5645] [General]
Hi

Environment:   
Windows - 10.0.19041.0 Microsoft Windows NT 10.0.19041.0
Ubuntu - Ubuntu 20.04 LTS

I have installed WSL2

 In powershell (running as admin)  I set the default version of wsl 
wsl --set-default-version 2

I then set wls -l -v
This tells me the Ubuntu version I have uses WSL 1

I then try wsl --set-version- Ubuntu-20.04 2

I get the error  
'There is no distribution with the supplied name'

In my Ubuntu I use lsb_release -a to get the version it displays

Distributor ID: Ubuntu
Description:    Ubuntu 20.04 LTS
Release:        20.04
Codename:       focal

So I thought my version was Ubuntu-20.04

Here is a screen shot of what happens 

![screen](https://user-images.githubusercontent.com/1951795/88329643-2e719800-cd22-11ea-8f54-792a8b82a8ec.png)

I looked at the Ubuntu version in the Settings -> Apps -> Apps and Features

This had the version as  2004.2020.424.0  

However running wsl --set-version Ubuntu-2004.2020.424.0  2 resulted in the same error

**Comment by therealkenc** [comment]
> In powershell (running as admin) I set the default version of wsl

#5428 as an initial matter. Do not run `wsl.exe` as Administrator.

The syntax above is incorrect. Docs are [here](https://docs.microsoft.com/en-us/windows/wsl/install-win10#set-your-distribution-version-to-wsl-1-or-wsl-2). For your scenario, with distribution `Ubuntu`:

```
PS C:\> wsl.exe --set-version Ubuntu 2
```

**Reply by jusito** [reply]
To show a list of available targets use: `wsl.exe -l -v`.
For ubuntu 20.04 lts its ubuntu-20.04 instead of ubuntu and version is current wsl version.

**Reply by moneygrower** [reply]
this syntax solved my problem, thanks

**Reply by sajjad-salam** [reply]
There is no distribution with the supplied name.

**Reply by somino12345** [reply]
PS C:\WINDOWS\system32> wsl --set-version Ubuntu-20.04 2
For information on key differences with WSL 2 please visit https://aka.ms/wsl2
Conversion in progress, this may take a few minutes.
A virtual disk support provider for the specified file was not found.
Error code: Wsl/Service/0xc03a0014
PS C:\WINDOWS\system32>     this my error I cant fix

**Comment by donnrri** [comment]
For anyone having this issue .
1. After setting wsl --set-default-version 2
2. Uninstall your Unbuntu
3. Install from the store again

**Comment by donnrri** [comment]
Thanks for your your response.  I appreciate your quick response but I looked at the Doc page .
It quite cleary says

`wsl --set-version <distribution name> <versionNumber>`

So I don't see how my use of the command is incorrect. If it is can the Docs be updated to reflect the correct command please.  If the Docs are correct please re-open this ticket.

**Comment by therealkenc** [comment]
> So I don't see how my use of the command is incorrect.

Your first try was `wsl --set-distribution ubuntu 20.04 2` which is incorrect because it has three parameters not two.

Your second try was `wsl --set-distribution Ubuntu-20.04 2` which is incorrect because your distribution NAME is `Ubuntu`.

![image](https://user-images.githubusercontent.com/7049700/88397565-f8133780-cd78-11ea-9979-c0f2da66ab9f.png)

**Comment by dinaayoub** [comment]
Using "wsl --set-distribution ubuntu 2" worked for me. thanks @therealkenc.

**Reply by Jawad-Mufti** [reply]
> Using "wsl --set-distribution ubuntu 2" worked for me. thanks @therealkenc.

Thank you!

**Comment by rcrapo** [comment]
I've tried the command "wsl --set-distribution ubuntu 2" as administrator  (I've also tried it with "wsl.exe" and it didn't work) and it still gives me the response "There is no distribution with the supplied name."

I've followed the instructions listed here: https://docs.microsoft.com/en-us/windows/wsl/wsl2-kernel

Does anyone have any ideas?

**Comment by therealkenc** [comment]
> I've tried the command "wsl --set-distribution ubuntu 2"

It is `--set-version` not `--set-distribution`.

**Comment by adsk2050** [comment]
--set-version is also not working. I have tried:
-  wsl.exe --set-version ubuntu 20.04 2
-  wsl.exe --set-version Ubuntu 20.04 2
-  wsl.exe --set-version ubuntu 2
-  wsl.exe --set-version Ubuntu 2
-  wsl.exe --set-version 'Ubuntu 20.04' 2
-  wsl.exe --set-version 'Ubuntu 20.04 LTS' 2

**Reply by colorsu** [reply]
reinstall the wsl2 worked for me

**Comment by therealkenc** [comment]
`wsl.exe -l -v` will list your distribution names. "`Ubuntu 20.04`" is not a distribution name (Ubuntu-20.04 has a dash in the name). To the extent you have a distribution with the name `Ubuntu` (sic) listed with `wsl.exe -l -v`, **and** `wsl --set-version Ubuntu 2` is "also not working", that would not be the same issue as the OP. In the OP, `Ubuntu` was installed.

**Reply by nsteblay** [reply]
There is a bug here.  The fix is to set WSL 2 as the default before installing a Linux kernel.  Use "wsl --set-default-version 2" from Powershell running as administrator before installing the Linux distribution of choice from the Microsoft Store.

**Comment by therealkenc** [comment]
`wsl --set-default-version` does not require Admin rights.

**Comment by Yuozas** [comment]
### Here's how you do it, don't uninstall stuff. Pwease :)
![What to do](https://i.imgur.com/VC1cAhp.png)
### I have different ridiculous stuff built and it all works after changes.
![How it looks](https://i.imgur.com/g2pED1w.png)

**Reply by AnshikaJaiswal28** [reply]
This totally worked for me thankyou!

**Comment by singhrajkr** [comment]
Try the below steps  : 

`PS C:\Users\rajku> **wsl --set-version Ubuntu-20.04 2**
Conversion in progress, this may take a few minutes...
For information on key differences with WSL 2 please visit https://aka.ms/wsl2
Conversion complete.
PS C:\Users\rajku> **wsl --list --all**
Windows Subsystem for Linux Distributions:
Ubuntu-20.04 (Default)
docker-desktop
docker-desktop-data
PS C:\Users\rajku> **wsl --list --verbose**
  NAME                   STATE           VERSION
* Ubuntu-20.04           Stopped         2
  docker-desktop         Stopped         2
  docker-desktop-data    Stopped         2`

Note: Replace **Ubuntu-20.04** with **Ubuntu-18.04** if you are running _Ubuntu-18.04_

**Comment by bahirbarak** [comment]
(Enable-WindowsOptionalFeature -Online -FeatureName Microsoft-Windows-Subsystem-Linux ) run this command in Powershell with admin privileges, it worked for me

**Comment by kalsmic** [comment]
This worked for me ```wsl --set-default Ubuntu-20.04 2 ```

**Comment by Zitzeronion** [comment]
I could not make it work with ```wsl --set-default Ubuntu-20.04 2```, however ```wsl --set-default Ubuntu 2``` did the trick for me.

**Comment by wnim** [comment]
https://docs.microsoft.com/en-us/windows/wsl/install

The documentation literally says:
"For example, wsl --set-version Ubuntu-20.04 2"

Even though it is wrong and should be
wsl --set-version Ubuntu 2

**Reply by Biswa96** [reply]
The documentation is just an example. It'll obviously vary with installed distribution.

**Reply by wnim** [reply]
the example is wrong even for the distribution it uses as an example
